templates/class.mustache

```diff
@@ -30,10 +30,15 @@ export interface ResponseWithBody<S extends number, T> extends Response {
     body: T;
 }
 
+export interface QueryParameters { [param: string]: any };
+
 export interface CommonRequestOptions {
-  $queryParameters?: {[param: string]: any};
+  $queryParameters?: QueryParameters;
   $domain?: string;
   $path?: string | ((path: string) => string);
+  $retries?: number; // see: https://github.com/visionmedia/superagent/blob/master/docs/index.md#retrying-requests
+  $timeout?: number; // see: https://github.com/visionmedia/superagent/blob/master/docs/index.md#timeouts
+  $deadline?: number; // see: https://github.com/visionmedia/superagent/blob/master/docs/index.md#timeouts
```

Comment:
add support for `retries`, `timeout`, and `deadline` via the common request options